test/unit/math/test_ad.hpp

```diff
@@ -1447,6 +1447,109 @@ void expect_float_equal(const T1& x, const T2& y) {
+ * Scalar vars.
+ * @tparam VarMat A `var_value` with an inner type inheriting from `EigenBase`.
+ * @tparam ResultMatVar The resulting type of applying a function to `A_mv`.
+ * @tparam ResultVarMat The result type of applying a function to `A_vm`.
+ * @param A_mv A eigen type of vars.
+ * @param A_vm A eigen type of vars.
+ * @param A_mv_f The result of a function from applying `A_mv`.
+ * @param A_vm_f The result of a function from applying `A_vm`
+ */
+template <typename MatVar, typename VarMat, typename ResultMatVar,
+          typename ResultVarMat, require_eigen_t<ResultMatVar>* = nullptr>
+inline void test_matvar_gradient(MatVar&& A_mv, VarMat&& A_vm,
+                                 ResultMatVar&& A_mv_f, ResultVarMat&& A_vm_f) {
+  for (Eigen::Index i = 0; i < A_vm.size(); ++i) {
+    A_vm_f.adj()(i) = 1;
+    A_mv_f.adj()(i) = 1;
+    stan::math::grad();
+    expect_near_rel("var<Matrix> vs Matrix<var> result value", A_vm.val(),
+                    A_mv.val(), 1e-12);
+    expect_near_rel("var<Matrix> vs Matrix<var> result adjoints", A_vm.adj(),
+                    A_mv.adj(), 1e-12);
+    expect_near_rel("var<Matrix> vs Matrix<var> input value", A_vm_f.val(),
+                    A_mv_f.val(), 1e-12);
+    expect_near_rel("var<Matrix> vs Matrix<var> input adjoints", A_vm_f.adj(),
+                    A_mv_f.adj(), 1e-12);
+    stan::math::set_zero_all_adjoints();
+  }
+}
+
+/**
+ * Test that the sum of the jacobian for matrices of vars is equal for the
+ *  var matrix.
+ * @tparam MatVar An Eigen type inheriting from `EigenBase` that has
+ * Scalar vars.
+ * @tparam VarMat A `var_value` with an inner type inheriting from `EigenBase`.
+ * @tparam ResultMatVar The resulting type of applying a function to `A_mv`.
+ * @tparam ResultVarMat The result type of applying a function to `A_vm`.
+ * @param A_mv A eigen type of vars.
+ * @param A_vm A eigen type of vars.
+ * @param A_mv_f The result of a function from applying `A_mv`.
+ * @param A_vm_f The result of a function from applying `A_vm`
+ */
+template <typename MatVar, typename VarMat, typename ResultMatVar,
+          typename ResultVarMat>
+inline void test_matvar_sum_gradient(MatVar&& A_mv, VarMat&& A_vm,
+                                     ResultMatVar&& A_mv_f,
+                                     ResultVarMat&& A_vm_f) {
+  using stan::math::var;
+  var b_mv = sum(A_mv_f);
+  var b_vm = sum(A_vm_f);
+  var final_var = b_mv + b_vm;
+  stan::math::grad(final_var.vi_);
+  if (!stan::math::isfinite(final_var)) {
+    return;
+  }
+  expect_near_rel("var<Matrix> vs Matrix<var> sum intermediate value",
+                  b_vm.val(), b_mv.val(), 1e-12);
+  expect_near_rel("var<Matrix> vs Matrix<var> sum intermediate adjoints",
+                  b_vm.adj(), b_mv.adj(), 1e-12);
+  expect_near_rel("var<Matrix> vs Matrix<var> sum result value", A_vm.val(),
+                  A_mv.val(), 1e-12);
+  expect_near_rel("var<Matrix> vs Matrix<var> sum result adjoints", A_vm.adj(),
```

Comment:
Don't check results adjoints

Reply:
The actual final value is `var final_var = b_mv + b_vm;` and we run everything up from it `stan::math::grad(final_var.vi_);` so wouldn't it make sense to check propagates right?

Reply:
Oh, I guess here that's fine. Certainly not going to hurt.